packages/salesforcedx-sobjects-faux-generator/src/retriever/orgObjectRetriever.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection } from '@salesforce/core';
+import { SObjectDescribe } from '../describe';
+import { SObjectSelector } from '../describe/sObjectDescribe';
+import { DeclarationGenerator } from '../generator/declarationGenerator';
+import { nls } from '../messages';
+import {
+  SObject,
+  SObjectDefinition,
+  SObjectDefinitionRetriever,
+  SObjectRefreshOutput
+} from '../types';
+
+export class OrgObjectRetriever implements SObjectDefinitionRetriever {
+  private describer: SObjectDescribe;
+  private declGenerator: DeclarationGenerator;
+  private selector: SObjectSelector;
+
+  public constructor(connection: Connection, selector: SObjectSelector) {
+    this.describer = new SObjectDescribe(connection);
+    this.declGenerator = new DeclarationGenerator();
+    this.selector = selector;
+  }
+
+  public async retrieve(output: SObjectRefreshOutput): Promise<void> {
+    let sobjects: string[] = [];
+    try {
+      sobjects = await this.describer.describeGlobal(this.selector);
+    } catch (e) {
+      const err = JSON.parse(e);
+      output.setError(
+        nls.localize('failure_fetching_sobjects_list_text', err.message),
+        err.stack
+      );
+      return;
+    }
+
+    let fetchedSObjects: SObject[] = [];
+    try {
+      fetchedSObjects = await this.describer.fetchObjects(sobjects);
```

Comment:
Question: Is the `describer` here loading the complete metadata for all SObjects in memory here? is this OK?

Would it be possible to architect it more like a "stream"? (For example: have the Transformer consume a "batch" at a time from the Describer, and then calling each Generator with a batch).

Reply:
@dobladez That's a valid concern, and I like the idea of streaming. Right now I feel that can be done as an incremental improvement, but I'll take a closer look.